### PR TITLE
More lock fixes

### PR DESCRIPTION
**Pull request by rennergade**
Fixes a few lock race condition problems and eliminates pack struct on seek.

### CODE REVIEW

File: seattlelib/lind_fs_calls.py [hunk 1]
```diff
@@ -1714,8 +1726,15 @@ def _cleanup_pipe(fd):
...
   # without changing to re-entrant locks
   def _close_helper(fd):
+
     filedescriptortable = masterfiledescriptortable[CONST_CAGEID]
 
+    # don't close streams, which have an inode of 1
+    try:
+      if filedescriptortable[fd]['inode'] == STREAMINODE: return 0
+    except KeyError:
+      pass
```
Comment (jesings) [inline seattlelib/lind_fs_calls.py:1736]: Why do we want to let fds without inodes through?

Comment (jesings) [reply seattlelib/lind_fs_calls.py:1736]: I know this is moved from elsewhere, but still.

**Review by jesings — COMMENTED** [review]
I was a little worried about your nested try-finally blocks, but I tested it out on my own and they should still propagate errors correctly. This PR looks good other than my one comment.

File: seattlelib/lind_fs_calls.py (hunk 1 above)
Comment (rennergade) [reply seattlelib/lind_fs_calls.py:1736]: There are fds without inodes, for example pipes, we want this not to crash there so they can be closed later.

Comment (jesings) [reply seattlelib/lind_fs_calls.py:1736]: Ah I had forgotten this was handled in the helper. Everything seems in order.

**Pull request merged**